core/controllers/acl_decorators.py

```diff
@@ -72,6 +92,41 @@ def test_can_access(self, *args, **kwargs):
     return test_can_access
 
 
+def does_classroom_exist(handler):
+    """Decorator to check whether classroom exists.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function.
+    """
+
+    def test_does_classroom_exist(self, classroom_url_fragment, **kwargs):
+        """Checks if classroom url fragment provided is valid. If so, return
+        handler or else redirect to the correct classroom.
+
+        Args:
+            classroom_url_fragment: str. The classroom url fragment.
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            handler. function. The newly decorated function.
+        """
+        classroom = classroom_services.get_classroom_by_url_fragment(
+            classroom_url_fragment)
+
+        if not classroom:
+            redirect_based_on_return_type(
+                self, '/learn/%s' % constants.DEFAULT_CLASSROOM,
+                self.GET_HANDLER_ERROR_RETURN_TYPE)
```

Comment:
Will statements in the decorated handler that come after this one get executed (and should they be)? (I.e. do you need a return statement after this one?)

Reply:
Not it is not expected to execute. I have added a return. Done.